tensorflow/python/ops/math_ops.py

```diff
@@ -1105,11 +1105,14 @@ def div_no_nan(x, y, name=None):
 
   Returns:
     The element-wise value of the x divided by y.
+  
+  Raises:
+    TypeError: if x and y are not of the same dtype.
   """
 
   with ops.name_scope(name, "div_no_nan", [x, y]) as name:
     x = ops.convert_to_tensor(x, name="x")
-    y = ops.convert_to_tensor(y, name="y", dtype=x.dtype.base_dtype)
+    y = ops.convert_to_tensor(y, name="y")
```

Comment:
I don't like this change as it makes code which now does not raise exceptions raise exceptions, like tf.div_no_nan(tf.constant(1.0), 2)

Reply:
While the example you mention above passes without raising exception in the current code, the current code fails if you just swap the order or change the combination and the exceptions raised are inconsistent:

Change in order:
```python 
tf.div_no_nan(2, tf.constant(1.0))
ValueError: Tensor conversion requested dtype int32 for Tensor with dtype float32: <tf.Tensor 'Const_2:0' shape=() dtype=float32>
```

 Change in combination:
```python

tf.div_no_nan(tf.constant(1), 2.0)
TypeError: Expected int32, got 2.0 of type 'float' instead.

```

The minor change in the PR will unify all these exceptions and make it consistent.

Reply:
In the meanwhile I can work on the cases where we pass one tensor (of float) and one int or vice versa can be handled with some other logic without raising exceptions.

Reply:
Sadly raising exceptions for code which currently does not violates TF's backward compatibility policy so we can't accept this part of the change.

Reply:
but this IS an issue right? There are 3 different exceptions being raised for basically the same input:
`TypeError` and `ValueError` being raised with varying error messages, sometimes from inner segments of code which the end user need not necessarily be exposed to. For an end user in a production system, catching these exceptions will be cumbersome and can break things. 

As for the backward compatibility, the `TypeError` currently being "raised" will never get raised in any scenario so might as well remove it then?

Reply:
I agree we need to unify things. Happy to remove the raises annotation if it doesn't apply.

Reply:
Removed the exception.